commit 88644bb9fee591b2743a881923263bc28df4cded
Author: Johan Hedberg <johan.hedberg@nokia.com>
Date:   Wed Jan 19 12:06:48 2011 +0530

    Revert "Bluetooth: Update sec_level/auth_type for already existing connections"
    
    This reverts commit 045309820afe047920a50de25634dab46a1e851d. That
    commit is wrong for two reasons:
    
    - The conn->sec_level shouldn't be updated without performing
    authentication first (as it's supposed to represent the level of
    security that the existing connection has)
    
    - A higher auth_type value doesn't mean "more secure" like the commit
    seems to assume. E.g. dedicated bonding with MITM protection is 0x03
    whereas general bonding without MITM protection is 0x04. hci_conn_auth
    already takes care of updating conn->auth_type so hci_connect doesn't
    need to do it.
    
    Signed-off-by: Johan Hedberg <johan.hedberg@nokia.com>
    Acked-by: Marcel Holtmann <marcel@holtmann.org>
    Signed-off-by: Gustavo F. Padovan <padovan@profusion.mobi>

diff --git a/net/bluetooth/hci_conn.c b/net/bluetooth/hci_conn.c
index 6b90a4191734..65a3fb5678eb 100644
--- a/net/bluetooth/hci_conn.c
+++ b/net/bluetooth/hci_conn.c
@@ -382,11 +382,6 @@ struct hci_conn *hci_connect(struct hci_dev *hdev, int type, bdaddr_t *dst, __u8
 		acl->sec_level = sec_level;
 		acl->auth_type = auth_type;
 		hci_acl_connect(acl);
-	} else {
-		if (acl->sec_level < sec_level)
-			acl->sec_level = sec_level;
-		if (acl->auth_type < auth_type)
-			acl->auth_type = auth_type;
 	}
 
 	if (type == ACL_LINK)